docs/proposals/sysctl.md

```diff
@@ -0,0 +1,717 @@
+5. b/c no memory is involved, i.e. purely functional difference
+
+**Note**: for all sysctls marked as "kmem memcg" kernel memory accounting must be enabled in the container for proper isolation. This will not be the case for 1.4, but is planned for 1.5.
+
+### Classification
+
+From the previous analysis the following classification is derived:
+
+| sysctl                       | ns    | accounting | reclaim   | pre-requisites |
+| ---------------------------- | ----- | ---------- | --------- | -------------- |
+| kernel.shm*                  | pod   | container  | pod       | i 1)           |
+| kernel.msg*                  | pod   | container  | pod       | i + ii + iii   |
+| fs.mqueue.*                  | pod   | container  | pod       | i + ii + iii   |
+| kernel.sem                   | pod   | container  | pod       | i + ii + iii   |
+| net.core.somaxconn           | pod   | container  | container | i + ii + iv    |
+| net.*.tcp_wmem/rmem          | host  | container  | container | i + ii + iv    |
+| net.core.wmem/rmem_max       | host  | container  | container | i + ii + iv    |
+| net.ipv4.ip_local_port_range | pod   | n/a        | n/a       | -              |
+| net.ipv4.tcp_syncookies      | pod   | n/a        | n/a       | -              |
+| net.ipv4.tcp_max_syn_backlog | pod   | n/a        | n/a       | -              |
+
+Explanation:
+
+- ns: value is namespaced on this level
+- accounting: memory is accounted for against limits of this level
+- reclaim: in the worst case, memory resources fall-through to this level and are accounted for there until they get destroyed
+- pre-requisites:
+  1. pod level cgroups
+  2. kmem acconuting enabled in Kubernetes
+  3. kmem accounting fixes for ipc namespace in Kernel >= 4.5
+  4. reliable kernel tcp net buffer accounting, which probably means to wait for cgroups v2.
+
+Footnote:
+
+1. Pod level cgroups don't exist today and pages are already re-parented on container deletion in v1.3. So supporting pod level sysctls in v1.4 that are tracked by user space memcg is not introducing any regression.
+
+**Note**: with the exception of `kernel.shm*` all of the listed pod-level sysctls depend on kernel memory accounting to be enabled for proper resource isolation. This will not be the case for 1.4 by default, but is planned in 1.5.
+
+**Note**: all the ipc objects persist when the originating containers dies. Their resources (if kmem accounting is enabled) fall back to the parent cgroup. As long as there is no pod level memory cgroup, the parent will be the container runtime, e.g. the docker daemon or the RunC process. It is [planned with v1.5 to introduce a pod level memory cgroup](pod-resource-management.md#implementation-status) which will fix this problem.
```

Comment:
What is a safe mechanism to reclaim these kernel memory pages? Will lowering the limit of memcg result in reclamation of these pages?